client/my-sites/google-my-business/stats/chart.js

```diff
@@ -52,18 +59,32 @@ class GoogleMyBusinessStatsChart extends Component {
 				transformedData: this.transformData( nextProps.data ),
 			} );
 		}
+
+		if (
+			this.props.interval !== nextProps.interval ||
+			this.props.siteId !== nextProps.siteId ||
+			this.props.statType !== nextProps.statType
+		) {
+			nextProps.requestGoogleMyBusinessStats(
+				nextProps.siteId,
+				nextProps.statType,
+				nextProps.interval,
+				'total'
+			);
+		}
 	}
 
 	shouldComponentUpdate( nextProps ) {
-		//@TODO: Once the data comes from redux, re-evaluate the need for deep equal
-		return (
-			this.props.interval !== nextProps.interval || ! isEqual( this.props.data, nextProps.data )
-		);
+		return this.props.interval !== nextProps.interval || this.props.data !== nextProps.data;
 	}
```

Comment:
@yurynix mentioned that we don't need to perform those checks when data come from Redux.